### PR TITLE
feat(pgwire): expose SSL functionality via RW_SSL_CERT and RW_SSL_KEY

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Expose SSL functionality via RW_SSL_CERT and RW_SSL_KEY

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

- expose SSL functionality via RW_SSL_CERT and RW_SSL_KEY
- Users can configure SSL cert and key file location via setting environment variables RW_SSL_CERT and RW_SSL_KEY for frontend nodes.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
After this PR, SSL can work?

**Comment by chenzl25** [comment]
> After this PR, SSL can work?

With a certificate with a localhost sign, I used psql to test it and it works. For on-prem users, they need to configure these environment variables by themselves. However, for cloud users, we have a proxy to provide TLS/SSL functionality.

```
psql "postgresql://root:@localhost:4566/dev?sslmode=verify-full"
psql (14.8 (Homebrew), server 9.5.0)
SSL connection (protocol: TLSv1.3, cipher: TLS_AES_256_GCM_SHA384, bits: 256, compression: off)
Type "help" for help.

dev=>
```

**Comment by chenzl25** [comment]
cc @mikechesterwang Could you please take a look based on your experience on SSL proxy?

**Review by liurenjie1024 — APPROVED** [review]

File: src/utils/pgwire/src/pg_protocol.rs [hunk 1]
```diff
@@ -114,16 +114,21 @@ pub struct TlsConfig {
 
 impl TlsConfig {
     pub fn new_default() -> Self {
-        let cert = PathBuf::new().join("tests/ssl/demo.crt");
-        let key = PathBuf::new().join("tests/ssl/demo.key");
-        let path_to_cur_proj = PathBuf::new().join("src/utils/pgwire");
+        let cert = if let Ok(cert) = std::env::var("RW_SSL_CERT") {
+            PathBuf::from(cert)
+        } else {
+            PathBuf::new().join("src/utils/pgwire").join("tests/ssl/demo.crt")
```
Comment (BugenZhao) [inline src/utils/pgwire/src/pg_protocol.rs]: Should this be test-only? If so, should we instead pass `None` here if they're unset?

https://github.com/risingwavelabs/risingwave/blob/6286ddd2a4447bc0ab8608c81ad9480b2e115752/src/frontend/src/lib.rs#L161

Comment (chenzl25) [reply src/utils/pgwire/src/pg_protocol.rs]: According to #5775  @wangrunji0408 do you know whether we still use SSL in deterministic tests?

Comment (BugenZhao) [reply src/utils/pgwire/src/pg_protocol.rs]: If it's believe okay to use "demo" cert in production, then we may consider embedding the contents with `include_bytes` and create a temp file at runtime.

Comment (chenzl25) [reply src/utils/pgwire/src/pg_protocol.rs]: If we don't use it anymore, I will remove those test purpose codes.

Comment (wangrunji0408) [reply src/utils/pgwire/src/pg_protocol.rs]: > According to #5775 @wangrunji0408 do you know whether we still use SSL in deterministic tests?

I remember SSL is always disabled in deterministic tests.

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**